File: KSXML.Server/KSXML/KSXML/KSXmlRecord.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace KSXML
{
    public class KSXmlRecord
    {
        private readonly Dictionary<String, Object> _dataMap = new Dictionary<String, Object>();

        public void SetFieldValue(String field, Object value)
        {
            if (_dataMap.ContainsKey(field))
            {
                _dataMap[field] = value;
            }
            else
            {
                _dataMap.Add(field, value);
            }
        }

        public string ToXml()
        {
            var sbuilder = new StringBuilder();

            sbuilder.Append("<ksxmlrecord>");
            foreach (var keyValuePair in _dataMap)
            {
                sbuilder.AppendFormat("<{0}>{1}</{0}>", keyValuePair.Key, keyValuePair.Value);
            }
            sbuilder.Append("</ksxmlrecord>");

            return sbuilder.ToString();
        }
    }
}

File: KSXML.Client/MyPlayList/src/com/ksxml/source/KSXmlRecord.java
package com.ksxml.source;

import java.util.HashMap;

public class KSXmlRecord {

	private HashMap<String, Object> _dataMap = new HashMap<String, Object>();
	
	public KSXmlRecord() {
		
	}
	
	// ## get field value
	public void setValue(String field, Object value) {
		this._dataMap.put(field, value);
	}
	
	// ## set field value
	public Object getValue(String field) {
		return this._dataMap.get(field);
	}
}
